extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/StartupBuildSteps.java

```diff
@@ -45,6 +49,31 @@
     static final MethodDescriptor CONTEXTUAL_DESTROY = MethodDescriptor.ofMethod(Contextual.class,
             "destroy", void.class, Object.class, CreationalContext.class);
 
+    private static final Logger LOGGER = Logger.getLogger(StartupBuildSteps.class);
+
+    @BuildStep
+    AnnotationsTransformerBuildItem annotationTransformer() {
+        return new AnnotationsTransformerBuildItem(new AnnotationsTransformer() {
+
+            @Override
+            public boolean appliesTo(org.jboss.jandex.AnnotationTarget.Kind kind) {
+                return kind == org.jboss.jandex.AnnotationTarget.Kind.CLASS;
+            }
+
+            @Override
+            public void transform(TransformationContext context) {
+                if (context.isClass() && !BuiltinScope.isDeclaredOn(context.getTarget().asClass())) {
```

Comment:
I know it's somewhat far-fetched but should we care about non-built in scopes? Users can have custom ones, or there is `Transactional` one already present. Both cases will pass this check and try to add yet another scope on the bean.

Reply:
That's a good point. Although in this case I doubt that a `@Startup` bean would be annotated with `@Transactional` or a custom scope. Let's create a separate issue to handle this corretly. WDYT?

Reply:
Hm, the problem is that annotation transformers are registered before the final set of scopes is known...

Reply:
I've created https://github.com/quarkusio/quarkus/issues/7076